site/docs/custom-catalog.md

```diff
@@ -0,0 +1,134 @@
+# Custom Catalog Implementation
+
+It's possible to read an iceberg table either from an hdfs path or from a hive table. It's also possible to use a custom metastore in place of hive. The steps to do that are as follows.
+
+- [Custom TableOperations](#custom-table-operations-implementation)
+- [Custom Catalog](#custom-table-implementation)
+- [Custom IcebergSource](#custom-icebergsource)
+
+### Custom table operations implementation
+Extend `BaseMetastoreTableOperations` to provide implementation on how to read and write metadata
+
+Example:
+```java
+class CustomTableOperations extends BaseMetastoreTableOperations {
+  private String dbName;
+  private String tableName;
+  private Configuration conf;
+
+  protected CustomTableOperations(Configuration conf, String dbName, String tableName) {
+    super(conf);
+    this.conf = conf;
+    this.dbName = dbName;
+    this.tableName = tableName;
+  }
+
+  // The refresh method should provide implementation on how to get the metadata location
+  @Override
+  public TableMetadata refresh() {
```

Comment:
I think #362 is quite close. Shall we merge that first and update this one to use `doRefresh`?